auctions/tests/utils_tests.py

```diff
@@ -0,0 +1,21 @@
+import fudge
+
+from django.test import TestCase
+
+from ..utils import issue_state, github_client
+
+
+class IssueStateTest(TestCase):
+
+    def test_return_state(self):
+        url = 'https://github.com/codesy/codesy/issues/158'
+
+        fake_gh_client = (fudge.Fake(github_client)
+                          .returns_fake()
+                          .provides('get_repo')
+                          .returns_fake()
+                          .provides('get_issue')
+                          .returns_fake()
+                          .has_attr(state='open'))
+
+        self.assertEqual('open', issue_state(url, fake_gh_client))
```

Comment:
Not sure why I didn't get the email about this 6 days ago. 

The biggest issues here are what is returned from the various method calls. The docs aren't exactly clear and need to be updated. 

* `provides` doesn't allow you to dictate how it acts when called, it simply says it's there and could be called if it needs to be.
* `expects` says that this method is going to be called and allows you say dictate what args it's expected to be called with (`with_args`) and/or what to return when that happens (`returns` or `returns_fake`).
* additionally, `expects_call` says that the `Fake` itself is a method that expects to be called, and can then be chained with the same `with_args`, `returns` and `returns_fake` calls.
* `returns_fake` creates a new `Fake` instance. This is particularly important to remember when assigning a chained series to a variable. In the case above, the final `returns_fake` call after `get_issue` is the one being assigned to `fake_gh_client` not the original, so the instance you are injecting into the function call is not the one you expect.
* I'm not entirely sure what `fudge.Fake(github_client)` was trying to accomplish. The only arg that `Fake` accepts (others are deprecated) is a name, which is simply used to indicate to the developer which instance was involved in a failed test. There's no way of basing the `Fake` instance off of an existing class like you can with `Mock`'s `spec` arg.

This should be what you want.

```python
fake_gh_client = fudge.Fake()
(fake_gh_client
 .expects('get_repo').with_args('codesy/codesy').returns_fake()
 .expects('get_issue').with_args(158).returns_fake()
 .has_attr(state='open'))
```